src/System.Collections.Immutable/tests/ImmutableArrayTest.cs

```diff
@@ -20,239 +23,326 @@ public class ImmutableArrayTest : SimpleElementImmutablesTestBase
-        public void Clear()
+        public static IEnumerable<object[]> Int32EnumerableData()
         {
-            Assert.Equal(ImmutableArray<int>.Empty, ImmutableArray.Create<int>().Clear());
-            Assert.Equal(ImmutableArray<int>.Empty, ImmutableArray.Create<int>(1).Clear());
-            Assert.Equal(ImmutableArray<int>.Empty, ImmutableArray.Create<int>(1, 2, 3).Clear());
+            // TODO: Standardize over Create vs new[].
+            yield return new object[] { new int[0] };
+            yield return new object[] { new[] { 1 } };
+            yield return new object[] { Enumerable.Range(1, 3) };
+            yield return new object[] { Enumerable.Range(4, 4) };
+            yield return new object[] { ImmutableArray.Create(2, 3, 5) };
         }
 
-        [Fact]
-        public void CreateEmpty()
+        public static IEnumerable<object[]> StrongBoxedInt32EnumerableData()
         {
-            Assert.Equal(ImmutableArray<int>.Empty, ImmutableArray.Create<int>());
-            Assert.Equal(ImmutableArray<int>.Empty, ImmutableArray.Create<int>(new int[0]));
+            // Once https://github.com/xunit/assert.xunit/pull/5 comes into corefx, all the StrongBox stuff can be removed.
+
+            return Int32EnumerableData()
+                .Select(array => array[0])
+                .Cast<IEnumerable<int>>()
+                .Select(enumerable => new object[]
+                {
+                    new StrongBox<IEnumerable<int>>(enumerable)
+                });
         }
 
-        [Fact]
-        public void CreateFromEnumerable()
+        public static IEnumerable<object[]> StrongBoxedSpecialInt32ImmutableArrayData()
         {
-            Assert.Throws<ArgumentNullException>("items", () => ImmutableArray.CreateRange((IEnumerable<int>)null));
+            // Once https://github.com/xunit/assert.xunit/pull/5 comes into corefx, all the StrongBox stuff can be removed.
 
-            IEnumerable<int> source = new[] { 1, 2, 3 };
-            var array = ImmutableArray.CreateRange(source);
-            Assert.Equal(3, array.Length);
+            yield return new object[] { new StrongBox<IEnumerable<int>>(s_emptyDefault) };
+            yield return new object[] { new StrongBox<IEnumerable<int>>(s_empty) };
         }
 
-        [Fact]
-        public void CreateFromEmptyEnumerableReturnsSingleton()
+        [Theory]
+        [MemberData(nameof(Int32EnumerableData))]
+        public void Clear(IEnumerable<int> source)
         {
-            IEnumerable<int> emptySource1 = new int[0];
-            var immutable = ImmutableArray.CreateRange(emptySource1);
-
-            // This equality check returns true if the underlying arrays are the same instance.
-            Assert.Equal(s_empty, immutable);
+            Assert.True(s_empty == source.ToImmutableArray().Clear());
         }
 
         [Fact]
-        public void CreateRangeFromImmutableArrayWithSelector()
+        public void CreateEnumerableElementType()
```

Comment:
Added extra tests here to ensure `Create` creates an immutable array of collections for non-arrays.